widget/embedded/src/containers/Wallets/Wallets.tsx

```diff
@@ -67,7 +65,9 @@ function Main(props: PropsWithChildren<PropTypes>) {
           supportedChainNames,
           meta.isContractWallet
         );
-        setAccounts(data);
+        if (data.length) {
```

Comment:
The issue has been fixed, but the twice (or multiple times) calling wallet-details API when switching accounts is back.